File: antlr-js/HtmlChatListener.js
import antlr4 from 'antlr4';
import ChatLexer from './ChatLexer.js';
import ChatParser from './ChatParser.js';
import ChatListener  from './ChatListener.js';

export default class HtmlChatListener extends ChatListener {    
    constructor(res) {
		super();
		this.Res = res;
	}
    
    enterName(ctx) {          
        this.Res.write("<strong>");    
    }
    
    exitName(ctx) {      
        this.Res.write(ctx.WORD().getText());
        this.Res.write("</strong> ");
    }
    
    enterColor(ctx) {     
        var color = ctx.WORD().getText();             
        ctx.text = '<span style="color: ' + color + '">';
    }
    
    exitColor(ctx) {         
        ctx.text += ctx.message().text;    
        ctx.text += '</span>';
    }
    
    exitEmoticon(ctx) {      
        var emoticon = ctx.getText();        
        
        if(emoticon == ':-)' || emoticon == ':)')
        {        
            ctx.text = "🙂";
        }
        
        if(emoticon == ':-(' || emoticon == ':(')
        {          
            ctx.text = "🙁";
        }
    }
    
    exitMessage(ctx) {                
        var text = '';
    
        for (var index = 0; index <  ctx.children.length; index++ ) {
            if(ctx.children[index].text != null)
                text += ctx.children[index].text;
            else
                text += ctx.children[index].getText();
        }
    
        if(ctx.parentCtx instanceof ChatParser.LineContext == false)
        {
            ctx.text = text;        
        }    
        else
        {
            this.Res.write(text);
            this.Res.write("</p>");
        }
    }
    
    enterCommand(ctx) {          
        if(ctx.SAYS() != null)
            this.Res.write(ctx.SAYS().getText() + ':' + '<p>');
    
        if(ctx.SHOUTS() != null)
            this.Res.write(ctx.SHOUTS().getText() + ':' + '<p style="text-transform: uppercase">');
    }
}

File: antlr-python/HtmlChatListener.py
import sys
from antlr4 import *
from ChatParser import ChatParser
from ChatListener import ChatListener

class HtmlChatListener(ChatListener) :
    def __init__(self, output):
        self.output = output
        self.output.write('<html><head><meta charset="UTF-8"/></head><body>')

    def enterName(self, ctx:ChatParser.NameContext) :
        self.output.write("<strong>") 

    def exitName(self, ctx:ChatParser.NameContext) :
        self.output.write(ctx.WORD().getText()) 
        self.output.write("</strong> ") 

    def enterColor(self, ctx:ChatParser.ColorContext) :
        color = ctx.WORD().getText()
        ctx.text = '<span style="color: ' + color + '">'        

    def exitColor(self, ctx:ChatParser.ColorContext):         
        ctx.text += ctx.message().text
        ctx.text += '</span>'

    def exitEmoticon(self, ctx:ChatParser.EmoticonContext) : 
        emoticon = ctx.getText()

        if emoticon == ':-)' or emoticon == ':)' :
            ctx.text = "🙂"
    
        if emoticon == ':-(' or emoticon == ':(' :
            ctx.text = "🙁"

    def enterLink(self, ctx:ChatParser.LinkContext):
        ctx.text = '<a href="%s">%s</a>' % (ctx.TEXT()[1], (ctx.TEXT()[0]))

    def exitMessage(self, ctx:ChatParser.MessageContext):
        text = ''

        for child in ctx.children:
            if hasattr(child, 'text'):
                text += child.text
            else:
                text += child.getText()
        
        if isinstance(ctx.parentCtx, ChatParser.LineContext) is False:
            ctx.text = text
        else:    
            self.output.write(text)
            self.output.write("</p>") 

    def enterCommand(self, ctx:ChatParser.CommandContext):
        if ctx.SAYS() is not None :
            self.output.write(ctx.SAYS().getText() + ':' + '<p>')

        if ctx.SHOUTS() is not None :
            self.output.write(ctx.SHOUTS().getText() + ':' + '<p style="text-transform: uppercase">')    

    def exitChat(self, ctx:ChatParser.ChatContext):
        self.output.write("</body></html>")